### DISCUSSION TITLE
Fitting CoxPHFitter with data observed up to specific date

### DISCUSSION THREAD

**Discussion by jakubruszkowski** [CamDavidsonPilon/lifelines#1334] [Q&A]
Hi.

I really love survival_difference_at_fixed_point_in_time_test() function that helps to compare the two curves at specific times, rather than comparing the entire survival curves against each other. I wonder whether it is possible to easily fit CoxPHFitter (and parametric models) with data up to a specific time (automatically censoring data on longer observed participants). It would be especially helpful when one wants to test predictors of 1-/5-/10-years mortality using the same dataframe.

**Comment by CamDavidsonPilon** [comment]
Ah, I think I understand. Is it correct that currently you would need to do some pre-processing in the dataframe. Something like:

```
for limit in [1, 5, 10]:
   df_ = df.copy()
   df_[E] = df_[T] >= limit
   df_[T] = np.minimum(df_[T], limit)
   cph = CoxPHFitter().fit(T, E, df_)
   cph ...
```

or something like that?
